src/tracker/stats_tracker_storage/redis/mod.rs

```diff
@@ -0,0 +1,181 @@
+// Copyright (c) 2024 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use async_trait::async_trait;
+use fastcrypto::hash::*;
+
+use anyhow::Result;
+use iota_types::base_types::IotaAddress;
+use itertools::Itertools;
+use redis::aio::ConnectionManager;
+use script_manager::ScriptManager;
+use serde_json::Value;
+use serde_json_canonicalizer::to_string;
+
+use crate::config::GasStationStorageConfig;
+
+use super::{Aggregate, AggregateType, StatsTrackerStorage};
+
+mod script_manager;
+
+#[derive(Clone)]
+pub struct RedisStatsTrackerStorage {
+    conn_manager: ConnectionManager,
+    // String format of the sponsor address to avoid converting it to string multiple times.
+    pub sponsor_key: String,
+}
+
+impl RedisStatsTrackerStorage {
+    pub async fn new(redis_url: impl AsRef<str>, sponsor: impl AsRef<str>) -> Self {
+        let client = redis::Client::open(redis_url.as_ref()).unwrap();
+        let conn_manager = ConnectionManager::new(client).await.unwrap();
+        Self {
+            conn_manager,
+            sponsor_key: sponsor.as_ref().to_string(),
+        }
+    }
+
+    #[cfg(test)]
+    pub async fn new_localhost() -> RedisStatsTrackerStorage {
+        use crate::test_env::random_address;
+        let sponsor_key = random_address().to_string();
+        Self::new("redis://127.0.0.1:6379", sponsor_key).await
+    }
+}
+
+#[async_trait]
+impl StatsTrackerStorage for RedisStatsTrackerStorage {
+    async fn update_aggr(&self, key: &[(String, Value)], update: &Aggregate) -> Result<f64> {
+        let hash = generate_hash_from_key(key);
+        let key = format!("{}:{}:{}", update.name, update.aggr_type, hash);
+
+        match update.aggr_type {
+            AggregateType::Sum => {
+                let script = ScriptManager::increment_aggr_sum_script();
+                let mut conn = self.conn_manager.clone();
+                let new_value: f64 = script
```

Comment:
Hm, was wondering why we use a float here, Redis [INCRBY](https://redis.io/docs/latest/commands/incrby/) should return an integer.

> [Integer reply](https://redis.io/docs/latest/develop/reference/protocol-spec#integers): the value of the key after the increment"